tests/phpunit/includes/tests-pods-field-boolean.php

```diff
@@ -239,7 +239,139 @@ public function test_method_validate_exists() {
+		$this->assertTrue( $this->field->validate( '0', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'No', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Off', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'False', null, $options ) );
+
+		// Other
+		$this->assertTrue( $this->field->validate( '', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Foobar', null, $options ) );
+
+		$options = array(
+			'boolean_format_type' => 'checkbox',
+			'required'            => false,
+		);
+
+		// Empty values.
+		$this->assertTrue( $this->field->validate( true, null, $options ) );
+		$this->assertTrue( $this->field->validate( 1, null, $options ) );
+		$this->assertTrue( $this->field->validate( '1', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Yes', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'On', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'True', null, $options ) );
+
+		// Non empty values.
+		$this->assertTrue( $this->field->validate( false, null, $options ) );
+		$this->assertTrue( $this->field->validate( 0, null, $options ) );
+		$this->assertTrue( $this->field->validate( '0', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'No', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Off', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'False', null, $options ) );
+
+		// Other
+		$this->assertTrue( $this->field->validate( '', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Foobar', null, $options ) );
+
+
+		/**
+		 * Required checkbox.
+		 * Only non_empty values are valid since a required checkbox only has one value.
+		 */
+
+		$options = array(
+			'boolean_format_type' => 'checkbox',
+			'required'            => true,
+		);
+
+		// Empty values.
+		$this->assertTrue( $this->field->validate( true, null, $options ) );
+		$this->assertTrue( $this->field->validate( 1, null, $options ) );
+		$this->assertTrue( $this->field->validate( '1', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Yes', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'On', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'True', null, $options ) );
+
+		// Non empty values.
+		$this->assertNotTrue( $this->field->validate( false, null, $options ) );
+		$this->assertNotTrue( $this->field->validate( 0, null, $options ) );
+		$this->assertNotTrue( $this->field->validate( '0', null, $options ) );
+		$this->assertNotTrue( $this->field->validate( 'No', null, $options ) );
+		$this->assertNotTrue( $this->field->validate( 'Off', null, $options ) );
+		$this->assertNotTrue( $this->field->validate( 'False', null, $options ) );
+
+		// Other
+		$this->assertNotTrue( $this->field->validate( '', null, $options ) ); // Parses to 0.
+		$this->assertNotTrue( $this->field->validate( 'Foobar', null, $options ) ); // Parses to 0.
+	}
```

Comment:
```suggestion
		$this->assertFalse( $this->field->validate( false, null, $options ) );
		$this->assertFalse( $this->field->validate( 0, null, $options ) );
		$this->assertFalse( $this->field->validate( '0', null, $options ) );
		$this->assertFalse( $this->field->validate( 'No', null, $options ) );
		$this->assertFalse( $this->field->validate( 'Off', null, $options ) );
		$this->assertFalse( $this->field->validate( 'False', null, $options ) );

		// Other
		$this->assertFalse( $this->field->validate( '', null, $options ) ); // Parses to 0.
		$this->assertFalse( $this->field->validate( 'Foobar', null, $options ) ); // Parses to 0.
	}
```

Reply:
This won't work. Validate will return an array of error messages. Not false.